python/tests/expressions/test_expressions_base.py

```diff
@@ -841,3 +924,290 @@ def test_not_expression_binding(unbound_not_expression, expected_bound_expressio
+    bound_expression = base.Or(
+        base.Not(
+            base.BoundIn[str](
+                base.BoundReference[str](
+                    field=NestedField(field_id=1, name="foo", field_type=StringType(), required=False),
+                    accessor=Accessor(position=0, inner=None),
+                ),
+                {StringLiteral("foo"), StringLiteral("bar")},
+            )
+        ),
+        base.Not(
+            base.BoundIn[str](
+                base.BoundReference[str](
+                    field=NestedField(field_id=2, name="bar", field_type=StringType(), required=False),
+                    accessor=Accessor(position=1, inner=None),
+                ),
+                {StringLiteral("baz"), StringLiteral("qux")},
+            )
+        ),
+    )
+    visitor = FooBoundBooleanExpressionVisitor()
+    result = base.visit(bound_expression, visitor=visitor)
+    assert result == ["IN", "NOT", "IN", "NOT", "OR"]
+
+
+def test_bound_boolean_expression_visitor_equal_and_not_equal():
+    """Test visiting an And expression with a bound boolean expression visitor"""
+    bound_expression = base.And(
+        base.BoundNotEqualTo[str](
+            term=base.BoundReference(
+                field=NestedField(field_id=1, name="foo", field_type=StringType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            ),
+            literal=StringLiteral("foo"),
+        ),
+        base.BoundEqualTo[str](
+            term=base.BoundReference(
+                field=NestedField(field_id=2, name="bar", field_type=StringType(), required=False),
+                accessor=Accessor(position=1, inner=None),
+            ),
+            literal=StringLiteral("foo"),
+        ),
+    )
+    visitor = FooBoundBooleanExpressionVisitor()
+    result = base.visit(bound_expression, visitor=visitor)
+    assert result == ["NOT_EQUAL", "EQUAL", "AND"]
+
+
+def test_bound_boolean_expression_visitor_always_true():
+    """Test visiting an And expression with a bound boolean expression visitor that resolves to AlwaysTrue"""
+    bound_expression = base.Or(
+        base.BoundIsNaN[str](
+            term=base.BoundReference(
+                field=NestedField(field_id=1, name="foo", field_type=StringType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            )
+        ),
+        base.BoundNotNaN[str](
+            term=base.BoundReference(
+                field=NestedField(field_id=1, name="foo", field_type=StringType(), required=False),
+                accessor=Accessor(position=0, inner=None),
+            )
+        ),
+    )
+    visitor = FooBoundBooleanExpressionVisitor()
+    result = base.visit(bound_expression, visitor=visitor)
+    assert result == ["TRUE"]
+
+
+def test_bound_boolean_expression_visitor_always_false():
+    """Test visiting an And expression with a bound boolean expression visitor that resolves to AlwaysFalse"""
+    bound_expression = base.And(
```

Comment:
Why not test the false/true expressions directly?

Reply:
Agreed and udated!